pkg/test/deployment/helm.go

```diff
@@ -108,39 +108,38 @@ func HelmTemplate(deploymentName, namespace, chartDir, workDir, valuesFile strin
 		valuesFileString = fmt.Sprintf(" --values %s", valuesFile)
 	}
 
-	err := initCharts(chartDir, workDir)
-	if err != nil {
+	helmRepoDir := workDir + "/helmrepo"
+	chartBuildDir := workDir + "/charts"
+	if err := os.MkdirAll(helmRepoDir, os.ModePerm); err != nil {
+		return "", err
+	}
+	if err := os.MkdirAll(chartBuildDir, os.ModePerm); err != nil {
 		return "", err
 	}
 
-	return shell.Execute(
-		"helm template %s --name %s --namespace %s%s%s",
-		chartDir, deploymentName, namespace, valuesFileString, valuesString)
-}
-
-func initCharts(chartDir, workDir string) error {
 	// Initialize the helm (but do not install tiller).
-	if err := exec("helm init --client-only"); err != nil {
-		return err
+	if _, err := exec(fmt.Sprintf("helm --home %s init --client-only", helmRepoDir)); err != nil {
+		return "", err
 	}
 
-	// Create a dir for the packaged charts.
-	chartBuildDir, err := ioutil.TempDir(workDir, "helm-charts-")
-	if err != nil {
-		return err
+	// Adding cni dependency as a workaround for now.
+	if _, err := exec(fmt.Sprintf("helm --home %s repo add istio.io %s",
+		helmRepoDir, "https://raw.githubusercontent.com/istio/istio.io/master/static/charts")); err != nil {
```

Comment:
Can't we consume this from the local enlistment? Wouldn't this break testing things in the local repo?

Reply:
There is a new hard-coded dependency on this from https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/requirements.yaml#L60

Without this the following helm commands will fail. Istio package is built below from local workspace.